frame/babe/src/lib.rs

```diff
@@ -224,10 +225,16 @@ decl_storage! {
 		/// on block finalization. Querying this storage entry outside of block
 		/// execution context should always yield zero.
 		Lateness get(fn lateness): T::BlockNumber;
+
+		EpochConfig: BabeEpochConfiguration;
+		NextEpochConfig: Option<BabeEpochConfiguration>;
 	}
 	add_extra_genesis {
 		config(authorities): Vec<(AuthorityId, BabeAuthorityWeight)>;
-		build(|config| Module::<T>::initialize_authorities(&config.authorities))
+		config(epoch_config): BabeEpochConfiguration;
+		build(|config| Module::<T>::initialize_authorities_and_epoch_config(
+			&config.authorities, &config.epoch_config
```

Comment:
`BabeEpochConfiguration` is `Encode`/`Decode` but not `Serialize`/`Deserialize` which means I can't use it here. Is there a wrapper type that I could use that wraps encodable things as serializable things perhaps?

Reply:
Can't you add a derive of `Serialize`/`Deserialize` for it? It's a tuple of `u64` and a plain enum for `AllowedSlots`, should work.